libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp

```diff
@@ -599,7 +605,28 @@ void AP_RCProtocol_CRSF::process_link_stats_frame(const void* data)
         }
     }
 
-    _link_status.rf_mode = MIN(link->rf_mode, 7U);
+    // Define the max number of RFModes based on ELRS modes, which is larger than Crossfire
+    const uint8_t max_modes = (RFMode::RF_MODE_MAX_MODES - RFMode::CRSF_RF_MAX_MODES) - 1U; // Subtract 1 due to zero-indexing
+    _link_status.rf_mode = MIN(link->rf_mode, max_modes); // Cap to avoid memory spills in the conversion tables
+
+#if AP_OSD_CRSF_EXTENSIONS_ENABLED
+    // Populate the extra data items
+    if(link->uplink_status > 0) {
+        _link_status.rssi_dbm = rssi_dbm;
+        _link_status.tx_power = link->uplink_tx_power < sizeof(AP_RCProtocol_CRSF::tx_powers)
+                                    ? AP_RCProtocol_CRSF::tx_powers[link->uplink_tx_power]
+                                    : -1;
```

Comment:
```suggestion
        _link_status.tx_power = -1;
        if (link->uplink_tx_power < sizeof(AP_RCProtocol_CRSF::tx_powers))
            _link_status.tx_power = AP_RCProtocol_CRSF::tx_powers[link->uplink_tx_power];
        }
```

(or use an `else`; in either case the ternary operator here make things less readable)

Reply:
... you could consider doing the "set all fields to -1" as a first step and then filling valid values in with non-minus-one values, actually.

Reply:
Good suggestions! Thanks!